### ISSUE TITLE
[no-unused-vars] false positive when use imported type in JSX generic component

### ISSUE THREAD

**Issue by claneo** [typescript-eslint/typescript-eslint#141]
**Repro**

```JSON
{
  "rules": {
    "react/jsx-uses-vars": "error"
  },
  "plugins": [
    "react"
  ],
  "extends": "plugin:@typescript-eslint/recommended"
}
```

```TS
import React from 'react';
import { TypeA, TypeB } from './interface';

type TypeC = string;

class GenericComponent<P> extends React.Component {
    public internalProp: P;
}

export const a = <GenericComponent<TypeA> />;

export const b: TypeB = '';

export const c = <GenericComponent<TypeC> />;
```

**Expected Result**

no error

**Actual Result**

`2:10  warning  'TypeA' is defined but never used  @typescript-eslint/no-unused-vars`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.1.0` |
| `@typescript-eslint/parser`        | `1.1.0` |
| `TypeScript`                       | `3.2.4` |
| `ESLint`                           | `5.12.1` |
| `node`                             | `10.14.1` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
I think this is a duplicate of #111 
We don't support JSX because it's supported in eslint-plugin-react

Though need to double check the AST for your example.
